app/src/main/java/org/stepic/droid/ui/dialogs/LoadingProgressDialog.kt

```diff
@@ -0,0 +1,26 @@
+package org.stepic.droid.ui.dialogs
+
+import android.app.Dialog
+import android.content.Context
+import android.os.Bundle
+import android.support.annotation.StringRes
+import android.view.Window
+import kotlinx.android.synthetic.main.dialog_progress.*
+
+import org.stepic.droid.R
+
+
+class LoadingProgressDialog
+@JvmOverloads constructor(
+        context: Context,
+        @StringRes private val titleRes: Int = R.string.loading) : Dialog(context) {
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        requestWindowFeature(Window.FEATURE_NO_TITLE)
+        setContentView(R.layout.dialog_progress)
+        title.setText(titleRes)
+        message.setText(R.string.loading_message)
+        setCancelable(false)
```

Comment:
It is dangerous if `ProgressDialog` is taking a long time.

Reply:
Moreover, documentation says: 

> Sets whether this dialog is cancelable with the BACK key.

So, we can cancel it, when clicked outside. See [setCanceledOnTouchOutside](https://developer.android.com/reference/android/app/Dialog.html#setCanceledOnTouchOutside%28boolean%29)

Reply:
What about cancelable on touch outside?